core/core.go

```diff
@@ -72,6 +73,12 @@ func (core *Core) Setup(sdkSettings model.SdkSettings, deviceProperties model.De
 		roxyPath = roxOptions.RoxyURL()
 	}
 
+	validAPIKeyPattern := "^[a-f\\d]{24}$"
+	matched, err := regexp.Match(validAPIKeyPattern, []byte(sdkSettings.APIKey()))
+	if roxyPath == "" && (err != nil || !matched) {
+		panic("Invalid rollout apikey")
```

Comment:
on the server, rox.go we have a recover that writes to the log
won't that means it will catch this panic? did you try running with a wrong app key in the demo app and so an exception (panic) ?